src/prefect/client/orion.py

```diff
@@ -816,9 +816,11 @@ async def match_work_queues(
             )
             if not new_queues:
                 break
-            filtered_queues = list(
-                filter(lambda q: q.name.startswith(prefix), new_queues)
-            )
+            filtered_queues = []
+            for q in new_queues:
+                for p in prefixes:
+                    if q.name.startswith(p):
+                        filtered_queues.append(q)
```

Comment:
this looks good! I think it might be a good idea to convert this to a set or something similar to avoid duplicate matches?

Reply:
I thought about it, but that implies that `new_queues` is a misnomer and it doesn't in fact, just hold new queues.

I just extended the already present logic to handle several filters.
https://github.com/PrefectHQ/prefect/pull/7222/files#diff-ed60480b777e77bdeeced26ed733ea8d0da54e3da5a6dce53054258a1625b004L811-L815

Reply:
@anticorrelator following up, do you have further comments?

Reply:
nope, this looks good to me!

Reply:
@jmg-duarte Actually, taking a look at the failing test it does look like queues might be being matched multiple times--for example if we have two prefixes specified but a queue name matches against both, it'll be added twice to the list

Reply:
Alright, I'll make a set! Thanks for the feedback!